File: PhaseGate/CCValidator/CCValidatorJava/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter credit card number: ");
        String cardNumber = scanner.nextLine();
        
        if (!isValidLength(cardNumber)) {
            System.out.println("Invalid: Card number must be between 13 and 16 digits");
            return;
        }
        
        String cardType = getCardType(cardNumber);
        if (cardType.equals("Unknown")) {
            System.out.println("Invalid: Unrecognized card type");
            return;
        }
        
        boolean isValid = isValidNumber(cardNumber);
        
        System.out.println("Card Type: " + cardType);
        System.out.println("Validity: " + (isValid ? "Valid" : "Invalid"));
        
        scanner.close();
    }
    
    private static boolean isValidLength(String cardNumber) {
        int length = cardNumber.length();
        return length >= 13 && length <= 16;
    }
    
    private static String getCardType(String cardNumber) {
        if (cardNumber.startsWith("4")) {
            return "Visa";
        } else if (cardNumber.startsWith("5")) {
            return "MasterCard";
        } else if (cardNumber.startsWith("37")) {
            return "American Express";
        } else if (cardNumber.startsWith("6")) {
            return "Discover";
        }
        return "Unknown";
    }
    
    private static boolean isValidNumber(String cardNumber) {
        int sumEven = 0;
        int sumOdd = 0;   
        
        
        for (int i = cardNumber.length() - 1; i >= 0; i--) {
            int digit = Character.getNumericValue(cardNumber.charAt(i));
            
            if ((cardNumber.length() - i) % 2 == 1) {
                sumOdd += digit;
            } else {
                int doubledDigit = digit * 2;
                sumEven += doubledDigit > 9 ? (doubledDigit % 10 + 1) : doubledDigit;
            }
        }
        
       
        int totalSum = sumEven + sumOdd;
        
        
        return totalSum % 10 == 0;
    }
}

File: PhaseGate/CCValidator/CCValidatorPython/credit_card_validator.py
def is_valid_length(card_number: str) -> bool:
    length = len(card_number)
    return 13 <= length <= 16

def get_card_type(card_number: str) -> str:
    if card_number.startswith('4'):
        return "Visa"
    elif card_number.startswith('5'):
        return "MasterCard"
    elif card_number.startswith('37'):
        return "American Express"
    elif card_number.startswith('6'):
        return "Discover"
    return "Unknown"

def is_valid_number(card_number: str) -> bool:
    sum_even = 0 
    sum_odd = 0

    for i in range(len(card_number) - 1, -1, -1):
        digit = int(card_number[i])
        if (len(card_number) - i) % 2 == 1:
            sum_odd += digit
        else:
            doubled_digit = digit * 2
            sum_even += (doubled_digit % 10 + 1) if doubled_digit > 9 else doubled_digit

    total_sum = sum_even + sum_odd
    
    return total_sum % 10 == 0

def main():
   card_number = input("Enter credit card number: ").strip()
    
    if not card_number.isdigit():
        print("Invalid: Card number must contain only digits")
        return

    if not is_valid_length(card_number):
        print("Invalid: Card number must be between 13 and 16 digits")
        return

    card_type = get_card_type(card_number)
    if card_type == "Unknown":
        print("Invalid: Unrecognized card type")
        return
    
    is_valid = is_valid_number(card_number)
    
    
    print(f"Card Type: {card_type}")
    print(f"Validity: {'Valid' if is_valid else 'Invalid'}")


main()